Please mark broken on darwin

```

this derivation will be built:
  /nix/store/ynkdj7r87lfw7ysim3d9kird0pxnxpv2-gnushogi-1.4.2.drv
this path will be fetched (0.37 MiB download, 0.37 MiB unpacked):
  /nix/store/2a9cf6lmjbmkgyzkxdn3lpmjnlnm59r4-gnushogi-1.4.2.tar.gz
copying path '/nix/store/2a9cf6lmjbmkgyzkxdn3lpmjnlnm59r4-gnushogi-1.4.2.tar.gz' from 'https://cache.nixos.org'...
building '/nix/store/ynkdj7r87lfw7ysim3d9kird0pxnxpv2-gnushogi-1.4.2.drv'...
unpacking sources
unpacking source archive /nix/store/2a9cf6lmjbmkgyzkxdn3lpmjnlnm59r4-gnushogi-1.4.2.tar.gz
source root is gnushogi-1.4.2
setting SOURCE_DATE_EPOCH to timestamp 1392669019 of file gnushogi-1.4.2/win32/Makefile
patching sources
configuring
configure flags: --prefix=/nix/store/y3197j3amzgca4549780w9vdv4i58idw-gnushogi-1.4.2
configure: C compiler
checking for gcc... clang
checking whether the C compiler works... no
configure: error: in `/private/tmp/nix-build-gnushogi-1.4.2.drv-0/gnushogi-1.4.2':
configure: error: C compiler cannot create executables
See `config.log' for more details
error: builder for '/nix/store/ynkdj7r87lfw7ysim3d9kird0pxnxpv2-gnushogi-1.4.2.drv' failed with exit code 77;
```

--- a/pkgs/games/gnushogi/default.nix
+++ b/pkgs/games/gnushogi/default.nix
@@ -10,11 +10,19 @@ stdenv.mkDerivation rec {
     sha256 = "0a9bsl2nbnb138lq0h14jfc5xvz7hpb2bcsj4mjn6g1hcsl4ik0y";
   };
 
+  env.LDFLAGS = lib.optionalString (!stdenv.isDarwin) "-Wl,-z,muldefs";
+
+  # Makefile ignores errors, so the build may silently succeed erroneously
+  postBuild = ''
+    test -e gnushogi/gnushogi || { echo "ERROR: no binary produced"; exit 1; }
+  '';
+
   meta = with lib; {
     description = "GNU implementation of Shogi, also known as Japanese Chess";
     homepage = "https://www.gnu.org/software/gnushogi/";
     license = licenses.gpl3;
     maintainers = [ maintainers.ciil ];
     platforms = platforms.unix;
+    broken = stdenv.isDarwin; # darwin does not support -z muldefs
   };
 }

